openedx/core/djangoapps/catalog/utils.py

```diff
@@ -124,6 +125,61 @@ def get_program_types(name=None):
         return []
 
 
+def get_credit_pathways(site, pathway_id=None):
+    """Read pathways from the cache.
+
+    The cache is populated by a management command, cache_programs.
+
+    Arguments:
+        site (Site): django.contrib.sites.models object
+
+    Keyword Arguments:
+        pathway_id (string): id identifying a specific pathway to read from the cache.
+
+    Returns:
+        list of dict, representing pathways.
+        dict, if a specific program is requested.
```

Comment:
Isn't the current implementation always returning type list of dict?

Reply:
If pathway_id is specified it calls cache.get() which returns single dict, otherwise it calls cache.get_many() which returns a list of all dicts (representing pathways).
This may have caused confusion, but this should also say "credit pathway" instead of "program".

Reply:
👍